core/webidl.rs

```diff
@@ -82,6 +82,41 @@ pub enum WebIdlErrorKind {
   Other(Box<dyn std::error::Error + Send + Sync + 'static>),
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub enum Type {
+  Null,
+  Undefined,
+  Boolean,
+  Number,
+  String,
+  Symbol,
+  BigInt,
+  Object,
+}
+
+pub fn type_of<'a>(
+  scope: &mut HandleScope<'a>,
+  value: Local<'a, Value>,
+) -> Type {
+  if value.is_null() {
+    return Type::Null;
+  }
+
+  #[allow(clippy::wildcard_in_or_patterns)]
+  match value.type_of(scope).to_rust_string_lossy(scope).as_str() {
+    "undefined" => Type::Undefined,
+    "boolean" => Type::Boolean,
+    "number" => Type::Number,
+    "string" => Type::String,
+    "symbol" => Type::Symbol,
+    "bigint" => Type::BigInt,
+    // Per ES spec, typeof returns an implementation-defined value that is not any of the existing ones for
+    // uncallable non-standard exotic objects. Yet Type() which the Web IDL spec depends on returns Object for
+    // such cases. So treat the default case as an object.
```

Comment:
nits: Per current ES spec, it no longer returns an implementation-defined value.
https://github.com/tc39/ecma262/pull/1441

Reply:
ah interesting; i'll remove the comment then